File: employee-management-system/src/main/java/com/yk/employees/services/EmployeeService.java
package com.yk.employees.services;

import com.yk.employees.model.Employee;

import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployees();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployeeById(Long id, Employee employee);
}

File: employee-management-system-ui/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = 'http://localhost:8089/api/employees';

class EmployeeService {

    saveEmployee(employee) {
        return axios.post(EMPLOYEE_API_BASE_URL, employee);
    }

    getAllEmployees() {
        return axios.get(EMPLOYEE_API_BASE_URL);
    }

    deleteEmployee(id) {
        return axios.delete(EMPLOYEE_API_BASE_URL + "/" + id);
    }

    getEmployeeById(id) {
        return axios.get(EMPLOYEE_API_BASE_URL + "/" + id);
    }

    updateEmployeeById(id, employee) {
        return axios.put(EMPLOYEE_API_BASE_URL + "/" + id, employee);
    }

}
export default new EmployeeService();